include/drivers/cavs_ipc.h

```diff
@@ -6,36 +6,12 @@
 
 #include <kernel.h>
 #include <device.h>
+#include <drivers/inotice.h>
```

Comment:
Is there any need for a drivers/cavs_ipc.h as I would expect the inotice header to define the operations that each SOC IPC driver would use ?

Reply:
cavs_ipc is a more featureful interface, with support for async "DONE" notification on command completion.  The driver supports the inotice interface, which is a subsystem that can be implemented portably by any driver.  But also has its own API for the more complicated stuff.

Is that "needed"?  I don't know, but the hardware is there and it works and we have a driver and a test.  Seems like we should probably leave it in if someone wants to be creative with this device in the future.  You could use it for talking to the CSME if someone wants to evolve the protocol, etc...

Alternatively we could move the header back into the soc include directory so it doesn't confuse anyone.

Reply:
Oh, and to be clear there is no overhead: the functions that are common between the interfaces are identical, the calls from cavs_ipc that are unused get optimized out.  The ISR is identical regardless.